psycopg/psycopgmodule.c

```diff
@@ -109,12 +189,16 @@ psyco_connect(PyObject *self, PyObject *args, PyObject *keywds)
         conn = PyObject_CallFunction(factory, "si", dsn, async);
```

Comment:
Personally, if this wouldn't break (presumably existing) connection subclasses, I'd just pass `kwargs` down here and let `connection_init()` run the `psyco_parse_args()` step.  This would remove the need for `_connect()` function altogether and would let any connection class being instantiated directly w/o any actual need for `connection_factory` parameter, like:
```py
conn = psycopg2.connection(dsn, **kwargs)
replconn = psycopg2.ReplicationConnection(dsn, replication_type=REPLICATION_LOGICAL, **kwargs)
```